### ISSUE TITLE
Golang type override package name conflicts with argument name

### ISSUE THREAD

**Issue by inconshreveable** [kyleconroy/sqlc#595]
Repro:

foo.sql:
```
-- name: GetFooByID :one
SELECT * FROM foo WHERE id = $1;
```

sqlc.json:
```
    overrides: [
        {
          "go_type": "*github.com/example/id.ID",
          "column": "foo.id"
        },
    ]
```

The generated code will use `id` for the argument name to the `GetFooByID` function which will conflict with the import because they will both have the same name `id`. sqlc needs to detect this case and rename one of these either by aliasing the import or by renaming the function argument.

**Comment by Maxiride** [comment]
Seems to me sort of a duplicate of #177 which is likely to be solved with #463, correct me if I understood wrongly.

**Comment by inconshreveable** [comment]
indeed. closing as a duplicate
